packages/vaex-ml/vaex/ml/transformations.py

```diff
@@ -31,14 +31,15 @@ class Transformer(HasState):
     '''
     features = traitlets.List(traitlets.Unicode(), help=help_features).tag(ui='SelectMultiple')
 
-    def fit_transform(self, df):
+    def fit_transform(self, df, target=None):
         '''Fit and apply the transformer to the supplied DataFrame.
 
         :param df: A vaex DataFrame.
+        :param target: A vaex expression containing the target variable.
 
         :returns copy: A shallow copy of the DataFrame that includes the transformations.
         '''
-        self.fit(df=df)
+        self.fit(df=df, target=target)
```

Comment:
```suggestion
        if target is None:
            self.fit()
        else:
            self.fit(df=df, target=target)
```

Reply:
I was following the example set py `scikit-learn`. I prefer that to your suggestion. Since in veax all these objects contain the `.transform` method, it makes sense that thay subclass form the same base. 

Alternatively, one could implement a different base for predictors/estimators, but they would also have the same methods as a transformer would + `.predict`.  So I don't know what is the best choice here. For me following the `sklearn` style made sense.